Source/Renderer/DrawCommand.js

```diff
@@ -498,8 +520,18 @@ define([
      * @param {Context} context The renderer context in which to draw.
      * @param {PassState} [passState] The state for the current render pass.
      */
-    DrawCommand.prototype.execute = function(context, passState) {
-        context.draw(this, passState);
+     DrawCommand.prototype.execute = function(context, passState) {
+         var useTimerQuery = defined(this._timerQuery) && context.timerQuery;
```

Comment:
For the common case when there is no timer query, there is way too many `if` checks here.  Make the common case fast with just one check.